pkg/api/controller_test.go

```diff
@@ -320,6 +320,25 @@ func TestController_CreateRepositoryHandler(t *testing.T) {
 			t.Fatalf("expected error creating duplicate repo")
 		}
 	})
+
+	t.Run("create repo with wrong storage type", func(t *testing.T) {
+		_, err := deps.catalog.CreateRepository(ctx, "repo3", onBlock(deps, "foo1"), "main")
+		if err != nil {
+			t.Fatal(err)
+		}
```

Comment:
This part looks redundant. You only want to create a single repo with a conflicting storage namespace (as you did in line 329)

Reply:
I want first to create a repo with some storage type, and then create another repo with different storage type. Sounds Ok?

Reply:
Why do you need to create the first repo? Do you mean that you are trying to test both the valid and invalid cases, or you meant something else? 

The sub-test description says "create repo with conflicting storage type", so what I expect to see is an attempt to create a repository with an invalid storage namespace that fails. 

If you are trying to test both cases I would separate their test. that is, you can define another test case with the description "create repo with a storage type" but this is already tested in line 289.

Reply:
You are right, this test is not necessary so I took it down.